packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts

```diff
@@ -61,6 +61,38 @@ export function isAzureProject(azureSettings: AzureSolutionSettings | undefined)
   return azureSettings !== undefined && HostTypeOptionAzure.id === azureSettings.hostType;
 }
 
+export function isBotProject(azureSettings: AzureSolutionSettings | undefined): boolean {
+  return (
+    azureSettings !== undefined &&
+    (azureSettings.capabilities?.includes(BotOptionItem.id) ||
+      azureSettings.capabilities?.includes(MessageExtensionItem.id))
+  );
+}
+
+export interface IBotTroubleShootMessage {
+  troubleShootLink: string;
+  textForLogging: string;
+  textForMsgBox: string;
+  textForActionButton: string;
+}
+
+export function getBotTroubleShootMessages(
+  azureSettings: AzureSolutionSettings | undefined
+): IBotTroubleShootMessage {
```

Comment:
Could we simply return BotTroubleShootMessage | undefined for this method, so that we don't need to expose isBotProject() to external

Reply:
Good idea. It potentially will make it simpler.

Reply:
After a second thought, it seems we can't do this. In the `undefined` case, we can't get the msg.textForLogging so will break the logic in `deploy.ts`.